Automattic/jetpack
# Backup: Add backup storage UI

#### Changes proposed in this Pull Request:

- [x] Add site storage size and storage limit on backup admin page
- [x] Add progress bar with different colors based on % of storage usage
- [x] Adjust new layout to include the new storage UI
- [x] When storage usage is above 65%, show a CUT with add-on storage upgrade

#### What's next?
The following features will be address in a new PR:
- Add days of backups saved below the storage meter.
- Backups stopped related messages on main title and CUT.

#### Screenshots
| Description | Screenshot |
|---|---|
| Admin page | <img width="500" alt="image" src="https://user-images.githubusercontent.com/1488641/213733024-cb771c7c-3abc-4083-b77a-41ef57bef775.png">
| Normal (< 65% used storage) | <img src="https://user-images.githubusercontent.com/1488641/213374355-8c74fc03-7e68-4ccc-aeed-bc3448ba6426.png" width="500"> |
| Warning (65% - 80% used storage) | <img src="https://user-images.githubusercontent.com/1488641/213373897-ee64af55-6188-4890-a94c-31caa990526f.png" width="500"> | 
| Critical (80%-100% used storage) | <img src="https://user-images.githubusercontent.com/1488641/213732435-c61b3142-c7cb-4572-b950-a6cdb6ef8b7c.png" width="500"> |
| Full (> 100% used storage) | <img src="https://user-images.githubusercontent.com/1488641/213374019-bac94275-5a41-4838-aa28-6385bff5be6a.png" width="500"> |

#### Other information:

#### Jetpack product discussion


* peaFOp-wy-p2
* p1HpG7-jQY-p2

#### Does this pull request change what data or activity we track or use?


No

#### Unit testing:
You can run unit testing using the following command:
* JS: `jetpack test packages/backup js`
* PHP: `jetpack test packages/backup php`

#### Testing instructions:





* Spin up a Jurassic Ninja site with Jetpack Beta, using Backup branch `add/backup-storage-ui` with a backup plan like Jetpack Backup (1 GB or 10 GB).
* Navigate to Jetpack -> VautlPress Backup
* Once the first backup completes, you should see the main page of the plugin, including the storage meter.
* Given it is a new site, it will be ~0GB used and no upsell prompt should be displayed.

To try different scenarios for the storage meter, you could try to upload big files to your site or dispatch the events using [Redux DevTools (chrome extension)](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en). After installing it, follow these steps:
* Open the extension in Chrome
* In the dropdown in the top right, select `jetpack-backup-plugin`
* In the bottom, click on `>_` option to show the dispatcher.
![image](https://user-images.githubusercontent.com/1488641/213371014-0eaa92d6-d057-41ea-8d1f-2f38a26d026e.png)
* It should look like this:
<img src="https://user-images.githubusercontent.com/1488641/213371077-5db3dc19-35cb-4210-9075-71ea2ef236f0.png" width="250">

* Then in the dispatcher editor, you can set the used storage like this (the size is represented in bytes):
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240
    }
}
```

* So if you have a site with 10 GB, you can use the following sizes to test the different scenarios:
  * Normal (< 65%): 1073741824 (1GB)
  * Warning (65% - 80%): 7516192768 (7GB)
  * Critical (80%-100%): 9663676416 (9GB)
  * Full (> 100%): 10737418240 (10 GB)
